mlflow/utils/conda.py

```diff
@@ -84,18 +111,42 @@ def get_or_create_conda_env(conda_env_path, env_id=None):
             "Conda executable by setting the {1} environment variable "
             "to the path of the Conda executable".format(conda_path, MLFLOW_CONDA_HOME)
         )
+
+    try:
+        process.exec_cmd([conda_env_create_path, "--help"], throw_on_error=False)
+    except EnvironmentError:
+        raise ExecutionException(
+            "You have set the env variable {0}, but {1} does not exist or "
+            "it is not working properly. Note that {1} and the conda executable need to be "
+            "in the same conda environment. You can change the search path by"
```

Comment:
This exception message is a little confusing to me. Perhaps try to trigger the exception and see if it makes sense (see my comment below on adding test on failure scenarios).

Reply:
ok, I added the test